rfcs/054-id-resolver/README.md

```diff
@@ -0,0 +1,155 @@
+# RFC 054: An Authoritative ID Resolver
+
+## What?
+
+A Catalogue Pipeline stage (or preliminary step in the ID Minter) and Concepts API feature to 
+retrieve additional information about Concepts that are derived from external authorities.
+
+The additional information includes:
+
+* The type of the referent
+* Identifiers and types for any components of a composite type
+
+## Where?
+
+The Catalogue Pipeline stage belongs between the Transformers and the ID Minter.
+
+The Concepts API feature will be associated with the authoritative-concepts index.
+
+## Why is it Needed?
+
+This proposal resolves four problems.  The first three of these problems manifest as either duplicate or missing Concepts in the Concepts API.
+
+### Type Inference
+
+Currently, transformers infer the type of a Concept from its use in the input. This can lead to incorrect and
+inconsistent types being applied.
+
+In MARC, for example, `x00` fields contain "Personal Name" fields, so are expected to contain Concepts of type Person.
+Places can be found in `651` (Geographic Term) fields, etc.  In these cases, the Authoritative Identifier for the Concept 
+is extracted from the `$0` subfield.
+
+However, this is neither validated before reaching the pipeline, nor are these rules reliable.
+
+In practice, a `700` field may contain a Person, or it may contain a text attributed to the named person.  Which one it is
+can only be known by checking the identifier from `$0` against a known record. (see the Malmonides Problem, below)
+
+### LoC namespace
+
+MARC data from Sierra does not distinguish between LCSH and LCNames. They are both signalled by the use of `indicator2=0`.
+As with type, above, the namespace is inferred from the field.  In "Name" fields (`x00`, `x10`, `x11`), it is assumed to be
+LCNames. In "Term" fields (`648`, `650`, `651`), it is assumed to be LCSH.
+
+This is not reliable.  Many "Geographical Terms" are in fact Names of Places.
+
+I believe that this could be resolved by examining the first character of the identifier (n for LCNames, s for LCSH).
+
+### Compound Concepts
+
+Compound concepts in MARC data from Sierra bear an identifier referring to the whole concept, but the transformer
+also breaks them up into the individual constituent concepts.  However, the identifiers for these constituent concepts 
+are only "label-derived".  LoC and MeSH entries for compound concepts also contain the identifiers for the constituents,
+so a new stage could populate these properly.
```

Comment:
Think this section could do with expanding a bit

Reply:
Yes, and in expanding it, I have discovered that the MADS data is insufficient to fix this problem :(